src/App.tsx

```diff
@@ -15,49 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+type Goods = string[];
+
+type Props = {
+  sortField: string;
+  isSorted: boolean;
+};
+
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, isSorted }: Props,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isSorted) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [isSorted, setIsSorted] = useState<boolean>(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, isSorted });
+
+  const resetFunc = () => {
+    setSortField('');
+    setIsSorted(false);
+  };
+
+  const toggleSort = () => {
+    setIsSorted(!isSorted);
+  };
+
   return (
+
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET })}
+          onClick={() => {
+            setSortField(SORT_FIELD_ALPHABET);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+          onClick={() => {
+            setSortField(SORT_FIELD_LENGTH);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning',
+            { 'is-light': !isSorted })}
+          onClick={toggleSort}
         >
           Reverse
         </button>
 
         <button
           type="button"
-          className="button is-danger is-light"
+          className={cn('button', 'is-danger', 'is-light',
+            { 'is-hidden': !sortField })}
+          onClick={resetFunc}
         >
           Reset
         </button>
       </div>
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+      {visibleGoods.map(good => (
+        <ul key={good}>
+          <li data-cy="Good">{good}</li>
```

Comment:
you need to map `li` tags in ul
<img width="583" alt="image" src="https://github.com/mate-academy/react_list-of-goods-hooks/assets/70220667/306b15eb-60d9-4596-b69e-1d704c3a80dd">

Reply:
not fixed